### PR TITLE
Converted the Windows-dependent build.bat into a build.sh shell script.

### PR DESCRIPTION
**Pull request by jahorton**
Exchanges our build script for a platform-independent one.

### CODE REVIEW

File: web/source/build.sh [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+}
+
+# Definition of global compile constants
+WEB_OUTPUT="../output"
+EMBED_OUTPUT="../embedded"
+SOURCE="."
+
+readonly WEB_OUTPUT
+readonly EMBED_OUTPUT
+readonly SOURCE
```
Comment (darcywong00) [inline web/source/build.sh]: interesting - I've never used readonly before.
Maybe you can combine the readonly and assignment lines

File: web/source/build.sh [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+: ${CLOSURECOMPILERPATH:=../tools}
+: ${JAVA:=java}
+
+compiler=$CLOSURECOMPILERPATH/compiler.jar
+compilecmd="$JAVA -jar $compiler"
+
+if ! [ -f $compiler ];
+then
+    echo File $compiler does not exist:  have you set the environment variable \$CLOSURECOMPILERPATH?
+fi
```
Comment (darcywong00) [inline web/source/build.sh]: Script should die like what the bat file had: `exit /B 1`.

File: web/source/build.sh [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+    echo Append SMP extensions
+    cat $EMBED_OUTPUT/kmw-smpstring.js kmwtemp.js > $EMBED_OUTPUT/keyman.js
+    rm kmwtemp.js
+
+    echo Compiled embedded application saved as $EMBED_OUTPUT/keyman.js
+
+    # Update any changed resources
+
+    # echo Copy or update resources
+    cp -R $SOURCE/resources/ $EMBED_OUTPUT/ >/dev/null
```
Comment (darcywong00) [inline web/source/build.sh]: Use  `cp -Rf` to correspond to bat version of `xcopy /E /Y`

Same of other uses of cp

File: web/source/build.sh (hunk 1 above)
Comment (jahorton) [inline web/source/build.sh]: Every example shows them being two separate lines, and apparently that pattern can be prone to errors at times.

**Pull request merged**